File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
#정렬

def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=lambda x: x*3, reverse=True)
    #000' 이나 '00' 등이 나올 경우 '0'으로 줄여서 반환하기 위함
    #str->int->str 문제에서 숫자크니까 문자열로 return하라고 함
    return str(int(''.join(numbers)))

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*; 

class Solution {
        public String solution(int[] numbers) {

            String[] numStrings = Arrays.stream(numbers)
                    .mapToObj(String::valueOf)
                    .toArray(String[]::new);

            //내림차순
            Arrays.sort(numStrings, (o1, o2) -> (o2 + o1).compareTo(o1 + o2));

            String result = String.join("", numStrings);

            if (result.startsWith("0")) {
                return "0";
            }
            return result;
        }
    }
